### ISSUE TITLE
Adding a SpinEdit / NumericUpDown Control to WPF

### ISSUE THREAD

**Issue by DarkIrata** [dotnet/wpf#2096]
The NumericUpDown control is available in WinForms, and also should be in WPF. Especially since it is much easier to make custom styles for controls in WPF. 
It is annoying to rebuild that control everytime you need it and can't reuse older code. Sure there are ready to use libraries like MahApps, but it should never be a requirement for basic controls to add a dependency. Even worse when you use style libraries like MaterialDesignInXaml which restyle many basic controls. (Yes, i know that an self-made problem) 

I am suprised that there is no request issue open for this.
But yeah, would be happy to get that control.

**Comment by mdtauk** [comment]
WinUI is building a NumberBox control, which includes Spin/Up Down buttons.

The design teams should confer to ensure a consistent design, and share knowledge and code as much as possible.

**Comment by ranjeshj** [comment]
WinUI now has a [NumberBox control. ](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/number-box)

**Comment by EddyHaigh** [comment]
With the move to use the Fluent Design based of [WPFUI](https://github.com/lepoco/wpfui) would it be possible to take that version of the number box and integrate it into the core framework?

**Comment by EddyHaigh** [comment]
While the WPFUI already incorporates the WinUI-style NumberBox, there's still a need for a classic NumberBox design. Given the ongoing design work and the absence of a NumberBox, this seems like an ideal opportunity to leverage WPFUI's expertise to introduce this essential missing component.

**Comment by james1301** [comment]
@grubioe is this the right issue to follow for NumberBox support? Are there any solid plans for this?

It is the main control missing from the WPF library. Support for Mins, Max's, and the number of decimal places.

**Comment by grubioe** [comment]
I am no longer working on WPF.  @harshit7962 can comment

**Comment by miloush** [comment]
Not saying there shouldn't be a separate control, but note that you can retemplate the Slider control to cover many of the scenarios.

**Comment by james1301** [comment]
@miloush yeah and everybody can individually create anything, it is about having the standard controls and building blocks for any application.

Not everybody creating their own slightly different number controls, probably with bugs in as the logic to control the cursor for a number and decimal points etc is not immediately straight forward.

The arrows up and down, are optional. They are the easiest bit to add, fundamentally what is most complicated and shouldn't require a third party control to just enter numbers. Personally I wouldn't want to be showing the up and down, as it adds little value in the majority of situations, especially with decimals.
